modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.search.SearchRequest;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Creates a new {@link DeleteByQueryRequest} that uses scrolling and bulk requests to delete all documents matching
+ * the query. This can have performance as well as visibility implications.
+ *
+ * Delete-by-query now has the following semantics:
+ * <ul>
+ *     <li>it's <tt>non-atomic</tt>, a delete-by-query may fail at any time while some documents matching the query have already been
+ *     deleted</li>
+ *     <li>it's <tt>try-once</tt>, a delete-by-query may fail at any time and will not retry it's execution. All retry logic is left to
+ *     the user</li>
+ *     <li>it's <tt>syntactic sugar</tt>, a delete-by-query is equivalent to a scroll search and corresponding bulk-deletes by ID</li>
+ *     <li>it's executed on a <tt>point-in-time</tt> snapshot, a delete-by-query will only delete the documents that are visible at the
+ *     point in time the delete-by-query was started, equivalent to the scroll API</li>
+ *     <li>it's <tt>consistent</tt>, a delete-by-query will yield consistent results across all replicas of a shard</li>
+ *     <li>it's <tt>forward-compatible</tt>, a delete-by-query will only send IDs to the shards as deletes such that no queries are
+ *     stored in the transaction logs that might not be supported in the future.</li>
+ *     <li>it's results won't be visible until the index is refreshed.</li>
+ * </ul>
+ */
+public class DeleteByQueryRequest extends AbstractBulkByScrollRequest<DeleteByQueryRequest> {
+
+    public DeleteByQueryRequest() {
+    }
+
+    public DeleteByQueryRequest(SearchRequest search) {
+        super(search);
+    }
+
+    @Override
+    protected DeleteByQueryRequest self() {
+        return this;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException e = super.validate();
+        if (getSearchRequest().indices() == null || getSearchRequest().indices().length == 0) {
```

Comment:
These fell like the same validations as are in update-by-query's request. Maybe a common superclass is in order?